UI/window-basic-auto-config-test.cpp

```diff
@@ -262,89 +277,132 @@ void AutoConfigTestPage::TestBandwidthThread()
 
-	obs_service_update(service, service_settings);
-	obs_service_apply_encoder_settings(service, vencoder_settings,
-					   aencoder_settings);
+	for (auto &service : services) {
+		obs_service_update(service, service_settings);
+		obs_service_apply_encoder_settings(service, vencoder_settings,
+						   aencoder_settings);
+	}
 
 	/* -----------------------------------*/
 	/* create output                      */
 
-	const char *output_type = obs_service_get_output_type(service);
+	const char *output_type = obs_service_get_output_type(services.front());
 	if (!output_type)
 		output_type = "rtmp_output";
 
-	OBSOutputAutoRelease output =
-		obs_output_create(output_type, "test_stream", nullptr, nullptr);
-	obs_output_update(output, output_settings);
+	std::vector<OutputWrapper> outputs;
+	bool encoder_updated = false;
 
-	const char *audio_codec = obs_output_get_supported_audio_codecs(output);
+	for (auto &service : services) {
+		OBSOutputAutoRelease output = obs_output_create(
+			output_type, "test_stream", nullptr, nullptr);
+		obs_output_update(output, output_settings);
 
-	if (strcmp(audio_codec, "aac") != 0) {
-		const char *id = FindAudioEncoderFromCodec(audio_codec);
-		aencoder = obs_audio_encoder_create(id, "test_audio", nullptr,
-						    0, nullptr);
-	}
+		if (!encoder_updated) {
+			const char *audio_codec =
+				obs_output_get_supported_audio_codecs(output);
 
-	/* -----------------------------------*/
-	/* connect encoders/services/outputs  */
+			if (strcmp(audio_codec, "aac") != 0) {
+				const char *id =
+					FindAudioEncoderFromCodec(audio_codec);
+				aencoder = obs_audio_encoder_create(
+					id, "test_audio", nullptr, 0, nullptr);
+			}
 
-	obs_encoder_update(vencoder, vencoder_settings);
-	obs_encoder_update(aencoder, aencoder_settings);
-	obs_encoder_set_video(vencoder, obs_get_video());
-	obs_encoder_set_audio(aencoder, obs_get_audio());
+			/* -----------------------------------*/
+			/* connect encoders/services/outputs  */
 
-	obs_output_set_video_encoder(output, vencoder);
-	obs_output_set_audio_encoder(output, aencoder, 0);
-	obs_output_set_reconnect_settings(output, 0, 0);
+			obs_encoder_update(vencoder, vencoder_settings);
+			obs_encoder_update(aencoder, aencoder_settings);
+			obs_encoder_set_video(vencoder, obs_get_video());
+			obs_encoder_set_audio(aencoder, obs_get_audio());
+			encoder_updated = true;
+		}
 
-	obs_output_set_service(output, service);
+		obs_output_set_video_encoder(output, vencoder);
+		obs_output_set_audio_encoder(output, aencoder, 0);
+		obs_output_set_reconnect_settings(output, 0, 0);
+
+		obs_output_set_service(output, service);
+
+		outputs.emplace_back(OutputWrapper{std::move(output)});
+	}
 
 	/* -----------------------------------*/
 	/* connect signals                    */
 
-	auto on_started = [&]() {
-		unique_lock<mutex> lock(m);
-		connected = true;
-		stopped = false;
-		cv.notify_one();
-	};
+	function<void(struct OutputWrapper &)> on_started =
```

Comment:
Is there a reason why auto can't be used here?

Reply:
The function is eventually put into a struct which needs exact type definition.